enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,138 @@
+ * `run packagemanifest` - covered by [OSDK-1126](https://issues.redhat.com/browse/OSDK-1126)
+ * `run bundle` - currently being implemented, design document is covered by [run bundle enhancement](https://github.com/operator-framework/enhancements/blob/master/enhancements/run-bundle.md)
+
+#### Command Line Interface
+The new command line will look as follows:
+
+```
+operator-sdk run bundle-upgrade (alias: bu) <bundle-image> --from-index=<> --starting-csv=<> [--channel=<>] [--operator-namespace=<>] [--install-mode=<>]
+```
+
+* Positional arguments:
+    * `bundle-image`
+* Required flags:
+    * `--from-index` 
+    * `--starting-csv`
+* Optional flags:
+    * `--channel` 
+    * `--operator-namespace`
+    * `--install-mode`
+
+
+* `<bundle-image>` is a positional argument that specifies the bundle image that we want to upgrade to
+* `[--from-index=<>]` is a required flag, which  contains the operator we’re upgrading from
+* `--starting-csv=<>` is a required flag
+
+* `[--channel=]` is an optional flag that specifies the channel to upgrade from within a subscription (default channel used if subscription does not contain a channel)
+  the operator. It will default to the `KUBECONFIG` context.
+* `[--operator-namespace=]` is an optional flag that specifies the namespace to install the operator. It will default to the `KUBECONFIG` context. (see run bundle)
+* `[--install-mode=]` is an optional flag that specifies the
+  [OLM install mode](https://red.ht/2YK0ksD). (see run bundle)
+  * We support 3 modes:
+    * AllNamespaces
+    * OwnNamespace
+    * SingleNamespace=ns1
+      * NOTE: if ns1 is the same as the operator's namespace, a warning will be
+        printed
+  * When `--install-mode` is not supplied, the default precedence order is:
+    * `OperatorGroup`, if it already exists
+    * `AllNamespaces`, if the CSV supports it
+    * `OwnNamespace`, if the CSV supports it
+    * If the CSV does not support either of these, `--install-mode` flag
+      will be **required**
+
+#### Assumptions
+* OLM is already installed
+* No previous version of the operator is already running
+* An index image exists that has the from version of the operator bundle
+* New version of the operator exists in the cluster 
+
+#### Example UX Scenario
+
+```
+operator-sdk run bundle-upgrade <bundle-image> --from-index quay.io/operator-framework/upstream-opm-builder --starting-csv csv.v1 --channel alpha
+```
+
+ 1. Create pod using `--from-index`, set container command (entrypoint) to:
+    ```
+    /bin/bash -c ‘ \
+      /bin/mkdir -p /database && \
+      /bin/opm registry add   -d /database/index.db -b {.BundleImage} && \
+      /bin/opm registry serve -d /database/index.db’
+    ```
+ 1. If pod fails, clean it up, and error out with message of unsupported index image. Also capture the pod's logs for debugging.
+ 1. Create GRPC CatalogSource, pointing to <pod-name>.<namespace>:50051
```

Comment:
Markdown gets confused on the angle brackets. I think enclosing in backticks should fix it.

```suggestion
 1. Create GRPC CatalogSource, pointing to `<pod-name>.<namespace>:50051`
```